fenix/app/src/main/java/org/mozilla/fenix/compose/Menu.kt

```diff
@@ -62,14 +69,50 @@ private fun Menu(
                 .background(color = FirefoxTheme.colors.layer2)
                 .then(modifier),
         ) {
+            val hasCheckedItems = menuItems.any { it.isChecked }
+
             for (item in menuItems) {
+                val modifierCheckMark = if (hasCheckedItems) {
+                    Modifier.selectable(
+                        selected = item.isChecked,
+                        role = Role.Button,
+                        onClick = {
+                            onDismissRequest()
+                            item.onClick()
+                        },
+                    )
+                } else {
+                    Modifier
+                }
+
                 DropdownMenuItem(
-                    modifier = Modifier.testTag(item.testTag),
+                    modifier = Modifier
+                        .testTag(item.testTag)
+                        .align(alignment = Alignment.CenterHorizontally)
+                        .then(modifierCheckMark),
                     onClick = {
                         onDismissRequest()
                         item.onClick()
                     },
                 ) {
+                    if (hasCheckedItems) {
+                        if (item.isChecked) {
+                            Icon(
+                                painter = painterResource(id = R.drawable.mozac_ic_checkmark_24),
+                                modifier = Modifier
+                                    .padding(end = 12.dp)
```

Comment:
I am thinking ahead of our grid system where we will likely turn these spacing between items into a token for the grid system. Instead of adding the padding here could we add this using a `Spacer` to make the transition process in the future easier.

Reply:
Done